src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -136,7 +136,7 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
-                cms.ComputeSignature(cmsSigner);
+                cms.ComputeSignature(cmsSigner, false); // silent is false to ensure PIN prompts appear if CNG/CAPI requires it
```

Comment:
This should only be done when restore was done in `--interactive` mode, right? Otherwise it could cause CI machines to hang if a cert PIN popup comes up.

Reply:
I didn't think long enough about my message. It's not getting signed in restore, it's a separate comment. If the sign command doesn't have a `--interactive` command, I think it should be added to reduce risk on CI machines. If sign does have an interactive mode, I think it should be used here.

Reply:
@zivkan agreed, just pointing out that this is existing behavior when on the .NET Framework today. .NET Core flipped the default overload's meaning in this case.